crates/nu-command/src/database/commands/into_sqlite.rs

```diff
@@ -164,27 +164,33 @@ fn operate(
     let file_name: Spanned<String> = call.req(engine_state, stack, 0)?;
     let table_name: Option<Spanned<String>> = call.get_flag(engine_state, stack, "table-name")?;
     let table = Table::new(&file_name, table_name)?;
+    let ctrl_c = engine_state.ctrlc.clone();
 
-    match action(input, table, span) {
+    match action(input, table, span, ctrl_c) {
         Ok(val) => Ok(val.into_pipeline_data()),
         Err(e) => Err(e),
     }
 }
 
-fn action(input: PipelineData, table: Table, span: Span) -> Result<Value, ShellError> {
+fn action(
+    input: PipelineData,
+    table: Table,
+    span: Span,
+    ctrl_c: Option<Arc<AtomicBool>>,
+) -> Result<Value, ShellError> {
     match input {
         PipelineData::ListStream(list_stream, _) => {
-            insert_in_transaction(list_stream.stream, list_stream.ctrlc, span, table)
+            insert_in_transaction(list_stream.stream, span, table, ctrl_c)
         }
         PipelineData::Value(
             Value::List {
                 vals,
                 internal_span,
             },
             _,
-        ) => insert_in_transaction(vals.into_iter(), None, internal_span, table),
```

Comment:
Oof, reminds me that I wanted to replace this `Option<Arc<AtomicBool>>` that is pervasive across our signatures with a more useful type. Would save us from `None` as well.